logger.tf

```diff
@@ -0,0 +1,49 @@
+locals {
+  api_mgmt_name = var.apim_suffix == "" ? "cft-api-mgmt-${var.environment}" : "cft-api-mgmt-${var.apim_suffix}"
+  api_mgmt_resource_group = "cft-${var.environment}-network-rg"
+  api_mgmt_logger_name = "${local.api_mgmt_name}-logger"
```

Comment:
These may not be needed if you're able to get the api_management_logger_id value from azurerm_api_management_logger - see next comment.